### ISSUE TITLE
Crash in Millard2012EquilibriumMuscle after cloning Model

### ISSUE THREAD

**Issue by tgeijten** [opensim-org/opensim-core#2958]
Steps to reproduce:

1. Create an `OpenSim::Model` containing a `Millard2012EquilibriumMuscle`
2. From multiple threads, perform the following actions:
    1. Create a clone of the model using `Model( const Model& )`
    2. Call `initSystem()` on the cloned model
    3. Run a simulation

This causes the program to crash randomly, unless a mutex is hold during steps 2.i and 2.ii. Especially the requirement to hold a mutex until _after_ `initSystem()` is surprising. Models that use `Thelen2003Muscle` work fine without any mutex.

The current workaround is problematic, since it yields a significant performance penalty. In addition, the behavior is an indication of a potentially severe underlying issue with `Millard2012EquilibriumMuscle`.

**Comment by tgeijten** [comment]
I it possible update the default Simbody version in opensim-core, so that it includes Adam's fix?

**Comment by aymanhab** [comment]
There's a branch and PR with the fix, please test it to see if this
actually fixes the issue. I tagged you on the PR.

**Comment by tgeijten** [comment]
Yes, it fixes the problem! Thanks @aymanhab, and great job @adamkewley.
